File: reactclient/src/interfaces/Launch.ts
interface Launch {
  flight_Number: number;
  mission_Name: string;
  mission_id: string[];
  upcoming: boolean;
  launch_Year: string;
  launch_date_unix: number;
  launch_date_utc: string;
  launch_date_local: string;
  is_tentative: boolean;
  tentative_max_precision: string;
  tbd: boolean;
  launch_window: number;
  rocket: {
    rocket_Id: string;
    rocket_Name: string;
    rocket_Type: string;
    first_Stage: {
      cores: {
        core_serial: string;
        flight: number;
        block: number;
        reused: boolean;
        land_success: boolean;
        landing_type: string;
        landing_vehicle: string;
      }[];
    };
    second_Stage: {
      payloads: {
        payload_Id: string;
        norad_id: number[];
        payload_Type: string;
        payload_mass_kg: number;
        payload_mass_lbs: number;
      }[];
    };
  };
  ships: string[];
  telemetry: {
    flight_club: string;
  };
  launch_site: {
    site_id: string;
    site_name: string;
    site_name_long: string;
  };
  launch_success: boolean;
  links: {
    mission_patch: string;
    article_link: string;
    video_link: string;
  };
  details: string;
  static_fire_date_utc: string;
  static_fire_date_unix: number;
  timeline: {
    [key: string]: number;
  };
  crew: null | any;
  last_date_update: string;
  last_ll_launch_date: string;
  last_ll_update: string;
  last_wiki_launch_date: string;
  last_wiki_revision: string | null;
  last_wiki_update: string;
  launch_date_source: string | null;
}


export default Launch;

File: API/SpaceXLaunchService/SpaceXLaunchService.Domain/Entities/Launch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SpaceXLaunchService.Domain.Entities
{
    public record Launch
    {
        public int Flight_Number { get; init; }
        public string? Mission_Name { get; init; } = null!;
        public List<string> Mission_Id { get; init; } = null!;
        public bool? Upcoming { get; init; }
        public string Launch_Year { get; init; } = null!;
        public long? Launch_Date_Unix { get; init; }
        public DateTime? Launch_Date_Utc { get; init; }
        public DateTimeOffset Launch_Date_Local { get; init; }
        public bool? Is_Tentative { get; init; }
        public string Tentative_Max_Precision { get; init; } = null!;
        public bool? Tbd { get; init; }
        public int? Launch_Window { get; init; }
        public Rocket Rocket { get; init; } = null!;
        public List<string> Ships { get; init; } = null!;
        public Telemetry Telemetry { get; init; } = null!;
        public LaunchSite Launch_Site { get; init; } = null!;
        public bool? Launch_Success { get; init; }
        public LaunchFailureDetails Launch_Failure_Details { get; init; } = null!;
        public Links Links { get; init; } = null!;
        public string Details { get; init; } = null!;
        public DateTime? Static_Fire_Date_Utc { get; set; }
        public long? Static_Fire_Date_Unix { get; init; }
        public Timeline Timeline { get; init; } = null!;
        public object Crew { get; init; } = null!;
        public DateTime? Last_Date_Update { get; set; }
        public DateTime? Last_Ll_LaunchDate { get; set; }
        public DateTime? Last_Ll_Update { get; set; }
        public DateTime? Last_Wiki_Launch_Date { get; set; }
        public string? Last_Wiki_Revision { get; set; }
        public DateTime? Last_Wiki_Update { get; set; }
        public string? Launch_Date_Source { get; set; }
    }

}
